Added section for on-prem FW rules (#174)
bcgov/public-cloud-techdocs: docs/aws/LZA/design-build-deploy/working-with-ses-and-smtp.md

# Amazon SES: Prefer API + IAM roles; use SMTP only for legacy systems

> This document explains our recommended approach for sending email with Amazon Simple Email Service (SES) in our AWS Landing Zone, and why **SMTP credentials should be treated as a fallback**.

Reviewer: ```suggestion
This document describes how to send email with Amazon Simple Email Service (SES) in our AWS Landing Zone. It also explains why **SMTP credentials are a fallback option**, not the default.
```

Amazon SES supports two primary ways to send email:

- **Amazon SES API (recommended)**  
- **SES SMTP interface (legacy / compatibility option)**

## Recommendation

### Preferred: Use the SES API with IAM roles (no long-lived secrets)

Wherever possible, integrate applications with SES using the **SES API** (AWS SDK/CLI). This allows workloads to authenticate using **IAM roles** (including assumed roles), avoiding long-lived user credentials entirely.

!!! note "Important nuance"

>     Use of role assumption applies to the **SES API**, not SMTP. The SMTP interface does **not** use role-based authentication.

Reviewer: ```suggestion
    Role assumption works only with the **SES API**. The SMTP interface does **not** support role-based authentication.
```

### Use SMTP only for legacy systems that cannot use the SES API

> Some legacy software only supports SMTP. In those cases, SES SMTP is acceptable, but note:

Reviewer: ```suggestion
Some legacy software only supports SMTP. In those cases, SES SMTP is acceptable, with the following constraints:
```

> - SES SMTP credentials are **not the same** as AWS access keys.

Reviewer: ```suggestion
- SES SMTP credentials are **different** from AWS access keys.
```
